### PR TITLE
Code Summary Table

### PR DESCRIPTION
**Pull request by ps48**
### Description
Added code summary table with badges.
 
### Issues Resolved
1. Deleted old badges
2. Updated badges for code cov
3. Updated documentation link
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by davidcui1225 — DISMISSED** [review]
LGTM- please re-run the Link Checker workflow before merging ✌️

File: README.md [hunk 1]
```diff
@@ -17,56 +11,136 @@
...
+- [SQL CLI](https://github.com/opensearch-project/sql/tree/main/sql-cli)
+- [SQL JDBC](https://github.com/opensearch-project/sql/tree/main/sql-jdbc)
+- [SQL ODBC](https://github.com/opensearch-project/sql/tree/main/sql-odbc)
+- [Query Workbench](https://github.com/opensearch-project/sql/tree/main/workbench)
+
+## Code Summary
+
+### SQL Engine
+
+|                              |                                                                                                                                              |
```
Comment (joshuali925) [inline README.md:29]: are the headers left empty on purpose?

**Review by joshuali925 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ps48) [reply README.md:29]: yes, that's a dummy header for the table. Without these md won't render it as a table.

**Comment by ps48** [comment]
@davidcui1225 
> LGTM- please re-run the Link Checker workflow before merging ✌️
Link checker is fixed in https://github.com/opensearch-project/sql/pull/313

**Review by joshuali925 — APPROVED** [review]

**Review by davidcui1225 — APPROVED** [review]

**Pull request merged**